src/app/app-platform/ContentApp.cpp

```diff
@@ -94,6 +94,15 @@ void ContentApp::SendAppObserverCommand(chip::Controller::DeviceCommissioner * c
     ChipLogProgress(Controller, "Completed send of AppObserver command");
 }
 
+bool ContentApp::HasSupportedCluster(chip::ClusterId clusterId) const {
+    for (const auto& supportedCluster : mSupportedClusters) {
+        if (static_cast<int>(clusterId) == supportedCluster.clusterIdentifier) {
```

Comment:
The fact that you had to static_cast here should have been a hint that things are not right.

Reply:
As mentioned above the date originates as int32 in Java, which we have to compare to uint32 values in C++. So either way we have to do (implicit or explicit) type ambiguations one way or the other to correctly do the comparison. I updated the code using the uint32 types in the C++ SupportedCluster struct and the result is that we do several type ambiguations before storing the int32_t/jint as uint32_t instead of when comparing values here. 

The previous approach mentioned here requires an explicit static_cast to compile. The new approach does not, as it silently converts. In general I think it's good to be (forced to be) explicit about type casts/ambiguations.